pkg/controller/worker/actuator_delete.go

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gardener/gardener-extension-provider-kubevirt/pkg/kubevirt"
+
+	extensionscontroller "github.com/gardener/gardener/extensions/pkg/controller"
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Delete deletes the Worker related resources.
+func (a *actuator) Delete(ctx context.Context, worker *extensionsv1alpha1.Worker, cluster *extensionscontroller.Cluster) error {
+	if err := a.Actuator.Delete(ctx, worker, cluster); err != nil {
+		return fmt.Errorf("failed to delete worker resouces: %v", err)
```

Comment:
Please always use `errors.Wrap` from `"github.com/pkg/errors"` to wrap errors. Please fix this elsewhere as well. Also, please prefer "could not do something" as opposed to "failed to do something". This is the style that is used in rest of the code, and also in other provider extensions.